crates/lib3h/src/transport/mod.rs

```diff
@@ -64,56 +66,61 @@ pub mod tests {
         uri_B: &Url,
     ) {
         // Connect
-        let actual_uri_a = node_A.bind(uri_A).unwrap();
-        let actual_uri_b = node_B.bind(uri_B).unwrap();
-        let idAB = node_A.connect(&actual_uri_b).unwrap();
-        println!("actual_uri: {}, idAB: {}", actual_uri_b, idAB);
+        let _actual_bind_uri_a = node_A.bind(uri_A).unwrap();
+        let actual_bind_uri_b = node_B.bind(uri_B).unwrap();
+        let idAB = node_A.connect(&actual_bind_uri_b).unwrap();
+        println!("actual_bind_uri_b: {}, idAB: {}", actual_bind_uri_b, idAB);
 
-        println!("calling node A process 1");
         let (_did_work, _event_list) = node_A.process().unwrap();
-
-        println!("calling node B process 1");
         let (_did_work, _event_list) = node_B.process().unwrap();
 
         // Send A -> B
         let payload = [1, 2, 3, 4];
         node_A.send(&[&idAB], &payload).unwrap();
-        println!("calling node B process 2");
         let mut did_work = false;
         let mut event_list = Vec::new();
         for _x in 0..NUM_PROCESS_LOOPS {
             let (did_work_B, mut event_list_B) = node_B.process().unwrap();
-            // let (_did_work_A, _event_list_A) = node_A.process().unwrap();
+            let (_did_work_A, _event_list_A) = node_A.process().unwrap();
             event_list.append(&mut event_list_B);
             did_work |= did_work_B;
         }
         assert!(did_work);
-        assert_eq!(event_list.len(), 1);
-        let recv_event = event_list[0].clone();
+        assert!(event_list.len() >= 1);
+        let recv_event = event_list.last().unwrap().clone();
         let (recv_id, recv_payload) = match recv_event {
             TransportEvent::Received(a, b) => (a, b),
-            _ => panic!("Received wrong TransportEvent type"),
+            e => panic!("Received wrong TransportEvent type: {:?}", e),
         };
-        assert_eq!(actual_uri_a.as_str(), recv_id);
+        assert!(node_A.get_uri(idAB.as_str()).is_some());
+        assert!(node_B.get_uri(recv_id.as_str()).is_some());
+
         assert_eq!(payload, recv_payload.as_slice());
-        println!("calling node A process 2");
         let (_did_work, _event_list) = node_A.process().unwrap();
 
         // Send B -> A
         let payload = [4, 2, 1, 3];
         let id_list = node_B.connection_id_list().unwrap();
         let idBA = id_list[0].clone();
         node_B.send(&[&idBA], &payload).unwrap();
-        println!("calling node A process 3");
-        let (did_work, event_list) = node_A.process().unwrap();
+        did_work = false;
+        event_list.clear();
+        for _x in 0..NUM_PROCESS_LOOPS {
+            let (did_work_A, mut event_list_A) = node_A.process().unwrap();
+            let (_did_work_B, _event_list_B) = node_B.process().unwrap();
+            did_work |= did_work_A;
+            event_list.append(&mut event_list_A);
+        }
```

Comment:
It might be best to have infinite loop with a break condition and a timeout panic.

Reply:
I made a TODO about this for now right above the code block.